Holovibes/sources/core/icompute.cc

```diff
@@ -67,21 +71,25 @@ bool ICompute::update_time_transformation_size(const unsigned short size)
 {
     try
     {
-        // Updates the size of the GPU P acc buffer.
         auto frame_res = input_queue_.get_fd().get_frame_res();
+
+        // Updates the size of the GPU P acc buffer.
         time_transformation_env_.gpu_p_acc_buffer.resize(frame_res * size);
 
-        // Updates the buffers for the moments, which depends on time_transformation_size
-        moments_env_.f0_buffer.resize(size);
-        moments_env_.f1_buffer.resize(size);
-        moments_env_.f2_buffer.resize(size);
+        if (api::get_data_type() != RecordedDataType::MOMENTS)
+        {
+            // Updates the buffers for the moments, which depends on time_transformation_size
+            moments_env_.f0_buffer.resize(size);
+            moments_env_.f1_buffer.resize(size);
+            moments_env_.f2_buffer.resize(size);
 
-        moments_env_.moment0_buffer.resize(frame_res);
-        moments_env_.moment1_buffer.resize(frame_res);
-        moments_env_.moment2_buffer.resize(frame_res);
+            moments_env_.moment0_buffer.resize(frame_res);
+            moments_env_.moment1_buffer.resize(frame_res);
+            moments_env_.moment2_buffer.resize(frame_res);
```

Comment:
You can delete these lines since now the buffers will be allocated in `allocate_moments_buffer`

Reply:
Couldn't I just move the other lines of the function to `allocate_moments_buffers` ?

Reply:
Nope, my bad